src/main/java/de/wejago/hichi2influx/repository/InfluxDbRepository.java

```diff
@@ -1,50 +1,43 @@
 package de.wejago.hichi2influx.repository;
 
-import com.influxdb.client.InfluxDBClient;
-import com.influxdb.client.WriteApiBlocking;
-import com.influxdb.client.domain.WritePrecision;
+import com.influxdb.client.WriteApi;
 import com.influxdb.client.write.Point;
 import de.wejago.hichi2influx.config.InfluxDBConfig;
-import de.wejago.hichi2influx.dto.SensorEntry;
-import java.time.Instant;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.scheduling.annotation.EnableScheduling;
+import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Repository;
 
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
 @Repository
 @RequiredArgsConstructor
+@Slf4j
+@EnableScheduling
 public class InfluxDbRepository {
+    private static final int SCHEDULE_INTERVAL_PERSIST_TO_DB = 30000;
     private final InfluxDBConfig influxDBConfig;
+    private final Queue<Point> measurementPoints = new ConcurrentLinkedQueue<>();
 
-    private InfluxDBClient influxDBClient;
-
-    public Point writePoint(SensorEntry sensorEntry) {
-        updateConnectionIfNeeded();
-        WriteApiBlocking writeApi = influxDBClient.getWriteApiBlocking();
-
-        Point point = generateMeasurementPoint(sensorEntry);
-        writeApi.writePoint(point);
-
-        return point;
+    public void writePoint(Point point) {
+        measurementPoints.add(point);
     }
 
-    private void updateConnectionIfNeeded() {
-        //set the connection only if it is empty of not active
-        if(influxDBClient == null || !influxDBClient.ping()) {
-            influxDBClient = influxDBConfig.dbConnection();
+    @Scheduled(fixedRate = SCHEDULE_INTERVAL_PERSIST_TO_DB)
+    public void persistPointsToInflux() {
+        if (!measurementPoints.isEmpty()) {
+            WriteApi writeApi = influxDBConfig.getWriteApi();
+            if (writeApi != null) {
+                int numberOfPointsWritten = 0;
+                while (!measurementPoints.isEmpty()) {
+                    Point point = measurementPoints.poll();
+                    writeApi.writePoint(point);
+                    numberOfPointsWritten++;
+                }
+                log.info("writePoint() wrote : " + numberOfPointsWritten + " point(s).");
```

Comment:
do we need this log output?

Reply:
Perhaps, when it is more than one entry, because then a previous write failed.

Reply:
disagree - then you will see, that more than one point was wrote, but this is not really valuable information. And in the regular case you will see only `wrote 1 point(s)` and this is not helpful.

Reply:
You see that there was something wrong before and it now reconnected and wrote X points.
Later, when it worked for some time, we can remove it.

Reply:
We did something in the middle - we will log X points only when X > 1.

Reply:
That's what I wrote ;-P